api/src/main/java/de/gaalop/dfg/Variable.java

```diff
@@ -48,14 +50,42 @@ public Variable(String name) {
 			this.name = name;
 		}
 	}
-	
+
+    public String getNewName(ControlFlowGraph graph, Boolean useArrays) {
+        if (this instanceof MultivectorComponent) {
+            MultivectorComponent component = (MultivectorComponent) this;
+
+            if (useArrays) {
+                // When using array, variables should be called e.g. var[6]
+                String name = getName() + "[" + component.getBladeIndex() + "]";
+                String bladeString = graph.getBladeString(component);
+//                System.out.println("Name = " + name + " # " + bladeString);
+                return name;
+            } else {
+                // Multivector variables should be called e.g. var_e12
+                String bladeString = graph.getBladeString(component);
+                bladeString = bladeString.replace(".0", ""); // Write var_1 instead of var_1.0
+                bladeString = removeExtraEs(bladeString); // Write var_e12 instead of var_e1e2

```

Comment:
I like the new handy syntax, but there might be a problem with high-dimensional geometric algebras for example G(12,2). Let's say the base vectors of G(12,2) are e1,e2,e3,...e9,e10,e11,e12. So the new blade name of e1^e2 is e12, but this is ambiguous with the blade name of base vector e12 which is also e12. How can we avoid the ambiguity?

Reply:
Thanks for taking your time and reviewing my changes! :) I'm glad you can welcome them.

The high-dimensional algebras are something I did not consider in the variable naming. It' a valid point. I think it's best if we use explicit naming like e1e2 as default. Maybe we could add a boolean to use the shortcut for lower dimensions but I actually don't know yet if that would be neccessary. I think I could implement this change in the next days.

I'd gladly document the new features at the Github wiki! Please feel free to complete the following list if I forgot something.
(General)
- Added #pragma as alternative to //#pragma
- pragma return
- pragma normalize
- pragma insert
(C# and Python only - here I'd need to add a new page I guess?)
- useArrays boolean and its effects
- optimize-on-save boolean and path

Reply:
Just to let you know, I added the changes of the variable naming + a small fix for Python.

Reply:
Hello Christian, may there be anything I can do so we can complete the pull request?